src/components/NormalRightRow/index.jsx

```diff
@@ -1,50 +1,81 @@
 import React from 'react';
 import { shape, string } from 'prop-types';
 import Typography from '@material-ui/core/Typography';
+import Chip from '@material-ui/core/Chip';
 import Tooltip from '../Tooltip';
-import { SKIP_KEYWORDS } from '../../utils/constants';
+import { SKIP_KEYWORDS, DESCRIPTOR_KEYWORDS } from '../../utils/constants';
 
 function NormalRightRow({ schema, classes }) {
   /**
    * Filter keywords that should be displayed in the right panel.
-   * (skip over keywords that are already displayed in other parts)
+   * (skip over keywords that do not need to be displayed)
    */
-  const keywords = Object.keys(schema).filter(
+  const specKeywords = Object.keys(schema).filter(
     key => !SKIP_KEYWORDS.includes(key)
   );
+  const descriptorKeywords = Object.keys(schema).filter(key =>
+    DESCRIPTOR_KEYWORDS.includes(key)
+  );
+
+  function displaySpecKeyword(keyword) {
+    if (
+      typeof schema[keyword] === 'object' &&
+      !Array.isArray(schema[keyword])
+    ) {
+      return <Tooltip key={keyword} keyword={keyword} classes={classes} />;
+    }
+
+    return (
+      <Chip
+        key={keyword}
+        label={`${keyword}: ${schema[keyword]}`}
+        size="small"
+        variant="outlined"
+      />
+    );
+  }
+
+  function displayDescriptor(keyword) {
+    return (
+      <Typography
+        key={keyword}
+        component="div"
+        variant="subtitle2"
+        className={classes.line}>
+        {`${keyword}: ${schema[keyword]}`}
+      </Typography>
+    );
+  }
+
+  function displayLines(specs, descriptors) {
+    const lines = [];
+
+    if (specs.length === 0 && descriptors.length === 0) {
+      lines.push(<div key="blank-line" className={classes.line} />);
+    }
+
+    if (specs.length > 0) {
+      lines.push(
+        <div key="spec-line" className={classes.line}>
+          {specs.map(keyword => displaySpecKeyword(keyword))}
+        </div>
+      );
+    }
+
+    if (descriptors.length > 0) {
+      if (specs.length > 0) {
+        lines.push(<div key="separator-line" className={classes.line} />);
+      }
+
+      descriptors.forEach(keyword => lines.push(displayDescriptor(keyword)));
+    }
+
+    return lines;
+  }
```

Comment:
A couple of things I noticed:
1. When there are more than 3 chips, a user is required to scroll the right panel to see the full list of chips.
2. If a description is really long, it will be really hard to read it currently since it requires scrolling horizontally bit by bit.
3. Looking for a given keyword chip requires scanning the list of chips from left to right.

Possible solutions:
1. Render at most 3 chips per line.
2. Add three triple dots at the end of the line to inform users that the description is longer than that. Upon hover, we could show the description in a tooltip.
3. Sort the list of chips alphabetically to improve the experience when looking for a certain keyword.

^ is something we can definitely do in a follow-up as an enhancement to the current implementation.

Reply:
Great idea! 👍 
Opened #51 to further implement these features!